### PR TITLE
2769 eliminate ts ignore in admin game setup component

### LINKED ISSUE #2769: Eliminate @ts-ignore in AdminGameSetup Component
## 🐞 Amendment description
The `AdminGameSetup` component currently uses `@ts-ignore` in several places. This is not a recommended practice as it suppresses TypeScript errors without providing a proper solution. We need to refactor the code to eliminate the usage of `@ts-ignore` and implement TypeScript-friendly solutions.

## 🖥️ Screenshots
393a98d96d2307b448cfedeecca32fa6

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
Fixes #2769 

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by Tristina1788** [comment]
@lilitkarapetyan @IanMayo I see the failed tests then I checkout and check from my side. I can run pass without any issue. So I re-run the job and the tests from deployment are also passed. The failed test before maybe from low connection. 
But I don't know what I should check in this PR. From UI, seems everything is working as before.

File: client/src/ActionsAndReducers/dbWargames/wargames_ActionCreators.ts [hunk 1]
```diff
@@ -155,18 +156,16 @@ export const populateWargameList = () => {
   return async (dispatch: WargameDispatch) => {
     dispatch(populatingDb(true))
 
-    // @ts-ignore
-    const wargameNames = await wargamesApi.populateWargameList(dispatch)
+    const wargameNames = await wargamesApi.populateWargameList()
     console.warn('now all wargameenams come from server', wargameNames)
```
Comment (IanMayo) [inline client/src/ActionsAndReducers/dbWargames/wargames_ActionCreators.ts:160]: @lilitkarapetyan - we can probably lose the above debug line

**Comment by IanMayo** [comment]
>  But I don't know what I should check in this PR. From UI, seems everything is working as before.

Hello @Tristina1788 - this is a code change that should not change the UI functionality.  So, everything should work as before - so that's a good sign :-)

**Comment by IanMayo** [comment]
We have these build errors @lilitkarapetyan.  I tried the apply the suggested fix, but I don't think it worked:
```
       src/ActionsAndReducers/dbMessageTypes/messageTypes_ActionCreators.ts(3,19): error TS7016: Could not find a declaration file for module 'check-types'. '/tmp/build_0e601687/client/node_modules/check-types/src/check-types.js' implicitly has an 'any' type.
         Try `npm i --save-dev @types/check-types` if it exists or add a new declaration (.d.ts) file containing `declare module 'check-types';`
       src/ActionsAndReducers/dbMessages/messages_ActionCreators.ts(4,19): error TS7016: Could not find a declaration file for module 'check-types'. '/tmp/build_0e601687/client/node_modules/check-types/src/check-types.js' implicitly has an 'any' type.
         Try `npm i --save-dev @types/check-types` if it exists or add a new declaration (.d.ts) file containing `declare module 'check-types';`
```

I'll have to leave it to you, please.

**Comment by lilitkarapetyan** [comment]
> We have these build errors @lilitkarapetyan. I tried the apply the suggested fix, but I don't think it worked:
> 
> ```
>        src/ActionsAndReducers/dbMessageTypes/messageTypes_ActionCreators.ts(3,19): error TS7016: Could not find a declaration file for module 'check-types'. '/tmp/build_0e601687/client/node_modules/check-types/src/check-types.js' implicitly has an 'any' type.
>          Try `npm i --save-dev @types/check-types` if it exists or add a new declaration (.d.ts) file containing `declare module 'check-types';`
>        src/ActionsAndReducers/dbMessages/messages_ActionCreators.ts(4,19): error TS7016: Could not find a declaration file for module 'check-types'. '/tmp/build_0e601687/client/node_modules/check-types/src/check-types.js' implicitly has an 'any' type.
>          Try `npm i --save-dev @types/check-types` if it exists or add a new declaration (.d.ts) file containing `declare module 'check-types';`
> ```
> 
> I'll have to leave it to you, please.

I will solove it

**Comment by lilitkarapetyan** [comment]
Did you run the yarn command ?

**Comment by IanMayo** [comment]
> Did you run the yarn command ?

The error comes from the Heroku build, not my local machine.  Running `yarn install` locally fixed my build, but we need the Heroku build to run, too.

**Review by IanMayo — DISMISSED** [review]

**Comment by IanMayo** [comment]
Hello @lilitkarapetyan - this is done, really. But, how about you try to remove some other `ts-ignore` comments in the code base, too?

**Comment by lilitkarapetyan** [comment]
> Hello @lilitkarapetyan - this is done, really. But, how about you try to remove some other `ts-ignore` comments in the code base, too?

I Think It's good Idea

**Comment by IanMayo** [comment]
This is good work, thanks @lilitkarapetyan 

It seems to cover quite a wide area, so it's worth @Tristina1788 doing some wider testing.  @lilitkarapetyan - could you please provide a list of areas of Serge that should be re-tested due to changes in this PR?

**Comment by lilitkarapetyan** [comment]
> This is good work, thanks @lilitkarapetyan
> 
> It seems to cover quite a wide area, so it's worth @Tristina1788 doing some wider testing. @lilitkarapetyan - could you please provide a list of areas of Serge that should be re-tested due to changes in this PR?

Hello , I think you could test keeping the Forces Save section in the admin panel and the Chanel tab in the game. 
Screenshot (303)
Screenshot (304)

**Comment by Tristina1788** [comment]
@IanMayo  @lilitkarapetyan Checking all places and I don't see the difference with before.

**Comment by IanMayo** [comment]
> I don't see the difference with before.

Good, that's what we're hoping to see 🤣   . Lilit has done some tidying (refactoring) of the source code for Serge, but the test is that nothing has been broken.  So, that's what we're looking for @Tristina1788 .

**Comment by lilitkarapetyan** [comment]
> @IanMayo @lilitkarapetyan Checking all places and I don't see the difference with before.

Yes, we need it to work like before.

**Comment by Tristina1788** [comment]
> > I don't see the difference with before.
> 
> Good, that's what we're hoping to see 🤣 . Lilit has done some tidying (refactoring) of the source code for Serge, but the test is that nothing has been broken. So, that's what we're looking for @Tristina1788 .

Yes, I understand that. So It's good to go.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**